engine/src/test/java/com/google/android/fhir/sync/remote/RemoteFhirServiceTest.kt

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.android.fhir.sync.remote
+
+import ca.uhn.fhir.context.FhirContext
+import com.google.android.fhir.logicalId
+import com.google.common.truth.Truth.assertThat
+import java.net.HttpURLConnection
+import kotlinx.coroutines.test.runTest
+import okhttp3.mockwebserver.MockResponse
+import okhttp3.mockwebserver.MockWebServer
+import org.hl7.fhir.r4.model.Bundle
+import org.hl7.fhir.r4.model.HumanName
+import org.hl7.fhir.r4.model.Patient
+import org.junit.After
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.robolectric.RobolectricTestRunner
+import retrofit2.Retrofit
+import retrofit2.converter.gson.GsonConverterFactory
+
+@RunWith(RobolectricTestRunner::class)
+class RemoteFhirServiceTest {
+  private val mockWebServer = MockWebServer()
+  private val apiService by lazy {
+    Retrofit.Builder()
+      .baseUrl(mockWebServer.url("/"))
+      .addConverterFactory(FhirConverterFactory.create())
+      .addConverterFactory(GsonConverterFactory.create())
+      .build()
+      .create(RemoteFhirService::class.java)
+  }
+  private val parser = FhirContext.forR4Cached().newJsonParser()
+
+  @Before
+  fun setUp() {
+    mockWebServer.start()
+  }
+
+  @After
+  fun shutdown() {
+    mockWebServer.shutdown()
+  }
+  @Test
+  fun download() = runTest {
```

Comment:
maybe separate this into multiple tests cases for @Body / gson converter / fhir converter.

```suggestion
  fun `should assemble request correctly`() = runTest {
    // This is to test that the @Body exists ...
    ...
    ... download()
    
    // there's no exception
  }
  
  fun `should use fhir converter .. ` () = runTest {
    ...
    // assert that the fhir patient has the right id...
  }
  
  ...
```

Reply:
Done. Also realised that `GsonConverterFactory` is not required. So removed it.